Yank/analyze.py

```diff
@@ -32,27 +35,1145 @@
+        measured from and how each phase interfaces with other phases.
+
+        Parameters
+        ----------
+        reporter : Reporter instance
+            Reporter from Repex which ties to the simulation data on disk.
+        name : str, Optional
+            Unique name you want to assign this phase, this is the name that will appear in MultiPhaseAnalyzer's. If not
+            set, it will be given the arbitrary name "phase#" where # is an integer, chosen in order that it is
+            assigned to the MultiPhaseAnalyzer.
+        reference_states: tuple of ints, length 2, Optional, Default: (0,-1)
+            Integers i and j of the state that is used for reference in observables, "O". These values are only used
+            when reporting single numbers or combining observables through MultiPhaseAnalyzer (since the number of states
+            between phases can be different). Calls to functions such as `get_free_energy` in a single Phase results in
+            the O being returned for all states.
+            For O completely defined by the state itself (i.e. no differences between states, e.g. Temperature)"
+                O[i] is returned
+                O[j] is not used
+            For O where differences between states are required (e.g. Free Energy):
+                O[i,j] = O[j] - O[i]
+        """
+        if not reporter.is_open():
+            reporter.open(mode='r')
+        self._reporter = reporter
+        observables = []
+        # Auto-determine the computable observables by inspection of non-flagged methods
+        # We determine valid observables by negation instead of just having each child implement the method to enforce
+        # uniform function naming conventions.
+        self._computed_observables = {}  # Cache of observables so the phase can be retrieved once computed
+        for observable in _ObservablesRegistry.observables():
+            if hasattr(self, "get_" + observable):
+                observables.append(observable)
+                self._computed_observables[observable] = None
+        # Cast observables to an immutable
+        self._observables = tuple(observables)
+        # Internal properties
+        self._name = name
+        # Start as default sign +, handle all sign conversion at peration time
+        self._sign = '+'
+        self._equilibration_data = None  # Internal tracker so the functions can get this data without recalculating it
+        # External properties
+        self._reference_states = None  # initialize the cache object
+        self.reference_states = reference_states
+        self._mbar = None
+        self._kT = None
+
+    @property
+    def name(self):
+        """User-readable name of the phase"""
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    @property
+    def observables(self):
```

Comment:
This does not return the `_ObervablesRegistry` object, it determines which observables it can compute from the registry, and stores that as an internal list. I do agree that we could hide this and use inspection in the `MultiPhase` analyzers to determine which are shared properties.